tests/rust-wasm-tests/src/dlmalloc.rs

```diff
@@ -0,0 +1,14 @@
+extern crate alloc;
+use alloc::vec::Vec;
+
+#[no_mangle]
+pub fn vec_alloc() {
+    let mut v = Vec::new();
+    if let Err(_) = v.try_reserve(4) {
+        unreachable!();
+    }
+    if let Err(_) = v.push_within_capacity(1) {
```

Comment:
@bitwalker I finished with no_std + dlmalloc example. `Vec::push` contains code to panic on OOM, which brings `call_indirect` into Wasm. I used unstable `Vec::push_with_capacity` instead.
I also found a `no_global_oom_handling` config option, which causes rustc not to compile methods in `alloc` that panic on OOM (`Vec::push`, `reserve`, etc.) We should use it to throw compilation errors if devs use those methods in their smart contract code. It works only with the unstable `-Z build-std=alloc,core` flag, which makes cargo to build std from the source.

Reply:
The code looks quite ugly from the smart contract dev point of view. We might prioritize `call_indirect` support to avoid it.

Reply:
I believe if you set `panic = "abort"` in the `Cargo.toml`, it should remove basically all panic-related code (it simply traps in those cases). This is typically the approach used in Wasm-compiled Rust projects, might be worth a shot to see if that helps.

In general though, I agree we'll want to have some kind of solution for indirect calls, we'll have to revisit when `pcall` is stabilized in Miden.

Reply:
Thanks for the tip! I did not know about `panic = "abort"`. It seems to be set to "abort" for wasm32 target by default since I did not notice any changes in the wasm code after I set it.
Actually, the `core::fmt` used for message formatting in panic calls caused all the `call_indirect` ops. I found `panic_immediate_abort` std feature in https://github.com/rust-lang/rust/blob/551c7183f8ae46c4b4fd0237acbdc937670bf74a/library/core/src/panicking.rs#L52-L73 and it eliminated all the `core::fmt` code. I switched back to using `Vec::push`.